src/main/java/org/opensearch/knn/index/codec/nativeindex/IndexBuildSetup.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.nativeindex;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+/**
+ * IndexBuildSetup encapsulates the configuration and parameters required for building an index.
+ * This includes the size of each vector, the dimensions of the vectors, and any quantization-related
+ * settings such as the output and state of quantization.
+ */
+@Getter
+@AllArgsConstructor
+final class IndexBuildSetup {
+    /**
+     * The number of bytes per vector.
+     */
+    private final int bytesPerVector;
+
+    /**
+     * The number of dimensions in the vector.
+     */
+    private final int dimensions;
```

Comment:
is this number of dimension in input vector or the quantized vector?

Reply:
No , It could be normal flow as well as quantized.

Reply:
In that case Java doc needs to be updated to reflect this.

Reply:
It's already says The number of dimensions in the vector. ....which is pretty generic here.....

Reply:
I think you miss the point, the question here is not about whether the java doc is generic or not. Its about will which dimension this variable represent. 
When you say this:
```
The number of dimensions in the vector.
```
it is unclear which vector you are taking about. Quantized vector/non quantized/ if its a binary vector type so is it representing dimension for vector which Faiss will take or length of byte array.

Reply:
I changed it to Dimension of Vector for Indexing